raiden-ts/src/transport/epics/messages.ts

```diff
@@ -123,26 +125,38 @@ export function matrixMessageGlobalSendEpic(
         );
         return EMPTY;
       }
-      const serverName = getServerName(matrix.getHomeserverUrl()),
-        roomAlias = `#${action.meta.roomName}:${serverName}`;
+      const serverName = getServerName(matrix.getHomeserverUrl());
+      const roomAlias = `#${action.meta.roomName}:${serverName}`;
+      const start = Date.now();
+      let retries = 0;
       return getRoom$(matrix, roomAlias).pipe(
         // send message!
-        mergeMap((room) => {
+        mergeMap(async (room) => {
+          retries++;
           return matrix.sendEvent(
             room.roomId,
             'm.room.message',
             { body: getMessageBody(action.payload.message), msgtype: 'm.text' },
             '',
           );
         }),
-        retryWhile(intervalFromConfig(config$), { maxRetries: 3, onErrors: [429, 500] }),
+        timeout(config.httpTimeout),
+        retryWhile(intervalFromConfig(config$), {
+          maxRetries: 3,
+          onErrors: [429, 500, 'Timeout', 'timeout'],
```

Comment:
Should we use `networkErrors` here instead of adding timeout matches manually?

Reply:
Will do.